hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/CollectionSizeNode.java

```diff
@@ -89,6 +90,9 @@ public String toSqlExpression() {
 			buffer.append( ownerKeyColumns[i] ).append( " = " ).append( collectionKeyColumns[i] );
 		}
 
+		String filterFragment = collectionDescriptor.filterFragment( collectionTableAlias,
```

Comment:
One minor quibble... either mark this var `final` or get rid of the var completely and just add the filter-fragment directly to the buffer.